src/coefficients.cpp

```diff
@@ -353,11 +355,13 @@ fk::matrix<P> generate_coefficients(
                 (-1);
           }
           trace_value_3 =
-              (legendre_poly_R_t * legendre_poly_R) * (+1 * flux_right / 2) +
+              (legendre_poly_R_t * legendre_poly_R) * (+1 * flux_right / 2) *
```

Comment:
Is `flux-right` type `P`? Mixing integer and floating point arithmetic is risky. 
```suggestion
              (legendre_poly_R_t * legendre_poly_R) * (0.5 * flux_right) *
```
If the change may repeated enough times in this file to justify a second PR.

Reply:
This should probably be pulled into a second PR. This happens all over the file.